### PR TITLE
AArch64: Initial version of ARM64Instruction and related files

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements the initial version of ARM64Instruction and
related files for aarch64.  This depends on #2660.

Signed-off-by: knn-k <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
Ready for review.

I am implementing ARM64BinaryEncoding.cpp and ARM64Instructions.cpp, but I am going to put them in another pull request.
In addition, some more ARM64*Instruction classes need to be defined.

File: compiler/aarch64/codegen/ARM64ConditionCode.hpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   ARMConditionCodeVC,
+   ARMConditionCodeHI,
+   ARMConditionCodeLS,
+   ARMConditionCodeGE,
+   ARMConditionCodeLT,
+   ARMConditionCodeGT,
+   ARMConditionCodeLE,
+   ARMConditionCodeAL,
+   ARMConditionCodeIllegal
+} ARM64ConditionCode;
```
Comment (knn-k) [inline compiler/aarch64/codegen/ARM64ConditionCode.hpp]: I have my version of compiler/aarch64/codegen/ARM64ConditionCode.hpp in #2704, which is not merged yet.  My version declares all the necessary symbols in its enum.  In addition, the type name is a little bit different: `TR_ARM64ConditionCode` vs `ARM64ConditionCode`.

File: compiler/aarch64/codegen/GenerateInstructions.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "compiler/aarch64/codegen/GenerateInstructions.hpp"
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/GenerateInstructions.cpp]: Does this path need the `compiler/` stem?

Comment (knn-k) [reply compiler/aarch64/codegen/GenerateInstructions.cpp]: It came from the p version of GenerateInstructions.cpp.  I am going to change it.

**Comment by knn-k** [comment]
I updated GenerateInstructions.cpp reflecting the review comment.
I also moved ARM64ConditionCode.hpp to another PR, #2744, because it conflicts with #2622.

**Comment by 0xdaryl** [comment]
Merging now because the AppVeyor build has no bearing on these changes.

**Pull request merged**